### DISCUSSION TITLE
Production deploy recipe

### DISCUSSION THREAD

**Discussion by linkinkov** [prisma/prisma#2460] [Q&A] [answered]
Hello everyone!
I use prisma 1 and now use prisma 2, how properly use new prisma at production?
After change in source code, need make prisma generate on production machine or just update sources? How organize artifact system without yarn install && prisma generate (if build of client need to be updated).

If possible, describe your experience in production.

**Comment by ryands17** [comment, marked as answer]
Hey @linkinkov :wave: 
For deploying an application that uses Prisma 2, you need to:

- run `prisma introspect` to generate the models in `schema.prisma` based on your models (here the URL of your production DB should be set in the `.env` file).
- run `prisma generate` to generate the types for your changed models and required targets so that is also available in the application.

Also if you're deploying to another system, like Ubuntu Linux or Serverless, you would have to add the required generators as well in `schema.prisma`.

For e.g. I am deploying this to AWS Lambda, so my `generator` block should contain the following:

```prisma
datasource pg {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}

generator client {
  provider      = "prisma-client-js"
  binaryTargets = ["native", "rhel-openssl-1.0.x"]
}

model User {
  id    Int     @default(autoincrement()) @id
  name  String?
  email String  @unique
}
```

So the main commands would be migrating and generating the client before deploying to production.

**Reply by linkinkov** [reply]
if i use 
`generator client {
  provider = "prisma-client-js"
}`
without binary target, on yarn install, it will install right binary?

**Reply by ryands17** [reply]
If the place you're running `yarn install` is the same place you will be deploying then it will work fine.

**Comment by ludralph** [comment]
Hi @linkinkov,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
